metrics/metricstest/resource_metrics_test.go

```diff
@@ -315,3 +319,53 @@ func TestMetricShortcuts(t *testing.T) {
 		}
 	}
 }
+
+func TestMetricFetch(t *testing.T) {
+	count := stats.Int64("count", "Test count metric", stats.UnitBytes)
+	tagKey := tag.MustNewKey("tag")
+	countView := &view.View{
+		Measure:     count,
+		Aggregation: view.Sum(),
+		TagKeys:     []tag.Key{tagKey},
+	}
+
+	view.Register(countView)
+
+	ctx, err := tag.New(context.Background(), tag.Upsert(tagKey, "alpha"))
+	if err != nil {
+		t.Errorf("Unable to create context: %v", err)
+	}
+	stats.Record(ctx, count.M(5))
+	stats.Record(ctx, count.M(3))
```

Comment:
Does the second `Record` ensures the first `Record` got recorded so we don't need to call `EnsureRecorded` here?

Reply:
Record pushes items into a `chan` of commands to a background goroutine managed by the Meter, but doesn't await a return message on the channel.

`EnsureRecorded` calls a function on the Meter that pushes a command onto the same `chan` and then waits for a response message on a return `chan`. Therefore, this acts as a barrier to ensure that all earlier commands on the `chan` have been processed, including all the previous `Record` calls. Since each Meter has its own background thread, we need to send a command on each (therefore the loop across all `metricproducers` on line 111 of `resource_metrics.go`).